tests/integration/test_allclose.py

```diff
@@ -0,0 +1,284 @@
+    b_np = np.array([x[1] for x in array_b]).reshape(shape_b)
+
+    a_num = num.array(a_np)
+    b_num = num.array(b_np)
+    res_np = np.allclose(a_np, b_np)
+    res_num = num.allclose(a_num, b_num)
+
+    assert res_np is True
+    assert res_np == res_num
+
+
+SHPAES_BROADCASTING2 = (
+    (1,),
+    (1, 1),
+    (1, 2, 1),
+    (2, 2, 1),
+)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize(
+    "shape_b", SHPAES_BROADCASTING2, ids=lambda shape_b: f"(shape_b={shape_b})"
+)
+def test_boradcast_true2(shape_b):
+    # for all cases,
+    # In Numpy, it pass
+    # In cuNumeric, it raises AttributeError:
+    # 'Store' object has no attribute '_broadcast'
+    shape_a = (3,)
+    size_a = np.prod(shape_a)
+    a_np = np.array(
+        [SCALARS_TRUE_DEFAULT[0][0] for _ in range(size_a)]
+    ).reshape(shape_a)
+
+    size_b = np.prod(shape_b)
+    b_np = np.array(
+        [SCALARS_TRUE_DEFAULT[0][1] for _ in range(size_b)]
+    ).reshape(shape_b)
+
+    a_num = num.array(a_np)
+    b_num = num.array(b_np)
+
+    res_np = np.allclose(a_np, b_np)
+    res_num = num.allclose(a_num, b_num)
+
+    assert res_np is True
+    assert res_np == res_num
+
+
+@pytest.mark.parametrize(
+    "equal_nan", (False, pytest.param(True, marks=pytest.mark.xfail))
+)
+@pytest.mark.parametrize(
+    "arr",
+    ([np.nan], [1, 2, np.nan], [[1, 2], [3, np.nan]]),
+    ids=lambda arr: f"(arr={arr})",
+)
+def test_equal_nan_basic(arr, equal_nan):
+    # If equal_nan is True,
+    # In Numpy, it pass
+    # In cuNumeric, it raises NotImplementedError
+    res_np = np.allclose(arr, arr, equal_nan=equal_nan)
+    res_num = num.allclose(arr, arr, equal_nan=equal_nan)
+    assert res_np == res_num
+
+
+EMPTY_ARRAY_PAIRS = (
+    ([], []),
+    ([], [[]]),
+    ([[]], [[]]),
+)
+
+
+@pytest.mark.parametrize(
+    "ab", EMPTY_ARRAY_PAIRS, ids=lambda ab: f"(a={ab[0]},b={ab[1]})"
+)
+def test_empty_array(ab):
+    a, b = ab
+    res_np = np.allclose(a, b)
+    res_num = num.allclose(a, b)
+
+    assert res_np is True
+    assert res_np == res_num
+
+
+SCALAR_BROADCASTING = (
+    (1e10, [1.00001e10]),
+    (1e10, [[1.00001e10]]),
+)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize(
+    "ab", SCALAR_BROADCASTING, ids=lambda ab: f"(a={ab[0]},b={ab[1]})"
+)
+def test_scalar_broadcasting(ab):
+    # for all cases,
+    # In Numpy, it pass
+    # In cuNumeric, it raises AttributeError:
+    # 'Store' object has no attribute '_broadcast'
+    a, b = ab
+    res_np = np.allclose(a, b)
+    res_num = num.allclose(a, b)
+
+    assert res_np is True
+    assert res_np == res_num
+
+
+@pytest.mark.parametrize(
+    "ab", SCALARS_FALSE_DEFAULT, ids=lambda ab: f"(a={ab[0]},b={ab[1]})"
+)
+def test_scalar_rtol_atol_true(ab):
+    a, b = ab
+    rtol = 1e-04
+    atol = 1e-06
+
+    res_np = np.allclose(a, b, rtol=rtol, atol=atol)
+    res_num = num.allclose(a, b, rtol=rtol, atol=atol)
+
+    assert res_np is True
+    assert res_np == res_num
```

Comment:
seems like there could be a `test_scalar_rtol_atol_false` that make the tolerances smaller so that cases true with the default tolerance become false

Reply:
The new case is added.